### PR TITLE
Refactor GW initialization

### PR DESCRIPTION
**Pull request by michalk8**

### CODE REVIEW

File: ott/core/gromov_wasserstein.py [hunk 1]
```diff
@@ -219,19 +225,21 @@ def cond_fn(
...
+      init = state.linear_state.q, state.linear_state.r, state.linear_state.g
       linear_pb = prob.update_lr_linearization(state.linear_state)
     else:
+      init = state.linear_state.f, state.linear_state.g
       linear_pb = prob.update_linearization(
           state.linear_state, solver.epsilon, state.old_transport_mass
       )
 
-    out = solver.linear_ot_solver(linear_pb)
+    out = solver.linear_ot_solver(linear_pb, init=init)
```
Comment (marcocuturi) [inline ott/core/gromov_wasserstein.py:242]: thanks for the catch

File: ott/core/quad_problems.py [hunk 2]
```diff
@@ -189,14 +193,17 @@ def is_low_rank(self) -> bool:
...
 
   @property
   def quad_loss(self) -> Tuple[Loss, Loss]:
+    """Quadratic part of the GW loss."""
     return self.loss.h1, self.loss.h2
 
   @property
   def is_balanced(self) -> bool:
+    """Whether the problem is balanced."""
     return ((not self.gw_unbalanced_correction) or
```
Comment (marcocuturi) [inline ott/core/quad_problems.py:207]: side comment: the fact that `gw_unbalanced_correction` is set to False seems to trigger the fact this is not balanced, regardless of `tau_a` and `tau_b`. Should this be a `and` ?

**Review by marcocuturi — APPROVED** [review]
Thanks for this Michal!

File: ott/core/quad_problems.py (hunk 2 above)
Comment (LaetitiaPapaxanthos) [reply ott/core/quad_problems.py:207]: If tau_a and tau_b are both equal to 1, then is_balanced return True, which is as expected. 

If tau_a and tau_b are not equal to 1, there are two cases. If there is no correction, `not self.gw_unbalanced_correction` is True, and in this case, it behaves as the balanced case (which is as expected as we ignore the unbalanced correction and the rescaling), if there is a correction, (not self.gw_unbalanced_correction is False) then the is_balanced function returns False. The logic is good I think.

**Comment by michalk8** [comment]
Looks like the broken tests (related to computing Hessians for Sinkhorn) are related to the new `jax` version (0.3.17), using `0.3.16` they pass.
Just tested this running `pytest -k "test_hessian_sinkhorn[True-0.93-0.91-shape6-0]"`, it's due to https://github.com/google/jax/pull/12041.

**Pull request merged**